python/src/spark_rapids_ml/umap.py

```diff
@@ -1201,36 +1338,62 @@ def _pre_process_data(
         return select_cols, multi_col_names, dimension, feature_type
 
 
-class UMAPModel(_CumlModel, UMAPClass, _UMAPCumlParams):
+class UMAPModel(_CumlModelWithColumns, UMAPClass, _UMAPCumlParams):
     def __init__(
         self,
         embedding_: List[pyspark.broadcast.Broadcast],
-        raw_data_: List[pyspark.broadcast.Broadcast],
+        raw_data_: Union[
+            List[pyspark.broadcast.Broadcast],
+            Dict[str, List[pyspark.broadcast.Broadcast]],
+        ],
+        sparse_fit: bool,
         n_cols: int,
         dtype: str,
     ) -> None:
         super(UMAPModel, self).__init__(
             embedding_=embedding_,
             raw_data_=raw_data_,
+            sparse_fit=sparse_fit,
             n_cols=n_cols,
             dtype=dtype,
         )
         self.embedding_ = embedding_
         self.raw_data_ = raw_data_
+        self._sparse_fit = sparse_fit  # If true, raw_data_ is a dict in CSR format
 
     @property
-    def embedding(self) -> List[List[float]]:
+    def embedding(self) -> np.ndarray:
+        """
+        Returns the model embeddings.
+        """
         res = []
         for chunk in self.embedding_:
-            res.extend(chunk.value.tolist())
-        return res
+            res.extend(chunk.value)
+        return np.array(res)  # TBD: return a more Spark-like object, e.g. DenseMatrix?
 
     @property
-    def raw_data(self) -> List[List[float]]:
-        res = []
-        for chunk in self.raw_data_:
-            res.extend(chunk.value.tolist())
-        return res
+    def rawData(self) -> Union[np.ndarray, scipy.sparse.csr_matrix]:
+        """
+        Returns the raw data used to fit the model. If the input data was sparse, this will be a scipy csr matrix.
+        """
+        if isinstance(self.raw_data_, dict):
+            indices = []
+            indptr = []
+            data = []
+            for i, chunk in enumerate(self.raw_data_["indices"]):
+                indices.extend(chunk.value)
+                indptr.extend(self.raw_data_["indptr"][i].value)
```

Comment:
there would be no need to take value here for example if underlying chunks weren't yet broadcast.